app.js

```diff
@@ -2,45 +2,68 @@ let readBtn = document.querySelector('button');
 let audio;
 let inputHTML = document.querySelector('input');
 let outputHTML = document.querySelector('p');
-let ones = ['', 'یەک', 'دوو', 'سێ', 'چوار', 'پێنج', 'شەش', 'حەوت', 'هەشت', 'نۆ'];
-let teens = ['دە', 'یانزە', 'دوانزە', 'سیانزە' ,'چواردە' ,'پانزە' ,'شانزە', 'حەڤە', 'هەژدە', 'نۆزدە'];
-let tens = ['','','بیست','سی','چل','پەنجا','شەست','حەفتا','هەشتا','نەوە'];
+let numbers = {
+    ones: ['', 'یەک', 'دوو', 'سێ', 'چوار', 'پێنج', 'شەش', 'حەوت', 'هەشت', 'نۆ'],
+    teens: ['دە', 'یانزە', 'دوانزە', 'سیانزە' ,'چواردە' ,'پانزە' ,'شانزە', 'حەڤە', 'هەژدە', 'نۆزدە'],
+    tens: ['','','بیست','سی','چل','پەنجا','شەست','حەفتا','هەشتا','نەوە']
+}
+
+inputHTML.addEventListener('input', (e) => {
+    outputHTML.innerHTML = "ئەنجام: " + convertToWords(e.target.value);
+    console.log(e);
+});
 
-inputHTML.addEventListener('input', () => {
-    outputHTML.innerHTML = "ئەنجام: " + convertToWords(inputHTML.value);
+readBtn.addEventListener('click', ()=> {
+    let number = inputHTML.value;
+    readNumber(number);
 });
 
 function playAudio(number) {
     audio = new Audio(`audio/${number}.m4a`);
     audio.play();
 }
-readBtn.addEventListener('click', ()=> {
-    let number = inputHTML.value;
-    readHundreds(number);
-    function readTens(number) {
-        if (number < 20) { 
-            playAudio(number);
-        } 
-        else if(number > 19) {
-            if(number % 10 != 0) {
-                playAudio(Math.floor(number / 10) * 10);
+function readNumber(number) {
+    if(number < 0) 
+        alert ('پشتگیری ژمارەی نێگەتڤ ناکات.');
+    else if(number.length > 3)
+        alert ('لە 3 ژمارە زیاتر ناخوێنێتەوە.');
+    else 
+        readHundreds(number);
+}
+function readTens(number) {
+    if (number < 20) { 
+        playAudio(number);
+    } 
+    else if(number > 19) {
+        if(number % 10 != 0) {
+            playAudio(Math.floor(number / 10) * 10);
+            audio.addEventListener('ended', ()=> {
+                playAudio('و');
                 audio.addEventListener('ended', ()=> {
+                    playAudio(number % 10);
+                })
+            })
```

Comment:
Registering event listeners every time this function is called is not ideal, you'll end up with hundreds of event listeners is user keep calling the function. Try exploring other approachs to implement this. See the answer about recursive approach here: https://stackoverflow.com/questions/54509959/how-do-i-play-audio-files-synchronously-in-javascript

Reply:
> you'll end up with hundreds of event listeners is user keep calling the function.

I couldn't understand what do you mean by this, would you mind explaining?